subsys/bluetooth/host/conn.c

```diff
@@ -130,6 +109,68 @@ static struct bt_conn sco_conns[CONFIG_BT_MAX_SCO_CONN];
 #endif /* CONFIG_BT_CLASSIC */
 #endif /* CONFIG_BT_CONN */
 
+#if defined(CONFIG_BT_CONN_TX)
+void frag_destroy(struct net_buf *buf);
+
+/* Storage for fragments (views) into the upper layers' PDUs. */
+/* TODO: remove user-data requirements */
+NET_BUF_POOL_FIXED_DEFINE(fragments, CONFIG_BT_CONN_FRAG_COUNT, 0,
+			  CONFIG_BT_CONN_TX_USER_DATA_SIZE, frag_destroy);
+
+struct frag_md {
+	struct bt_buf_view_meta view_meta;
+};
+struct frag_md frag_md_pool[CONFIG_BT_CONN_FRAG_COUNT];
+
+struct frag_md *get_frag_md(struct net_buf *fragment)
+{
+	return &frag_md_pool[net_buf_id(fragment)];
+}
+
+void frag_destroy(struct net_buf *frag)
+{
+	/* allow next view to be allocated (and unlock the parent buf) */
+	bt_buf_destroy_view(frag, &get_frag_md(frag)->view_meta);
+
+	LOG_DBG("");
+
+	/* Kick the TX processor to send the rest of the frags. */
+	bt_tx_irq_raise();
+}
+
+static struct net_buf *get_data_frag(struct net_buf *outside, size_t winsize)
+{
+	struct net_buf *window;
+
+	__ASSERT_NO_MSG(!bt_buf_has_view(outside));
+
+	/* Keeping a ref is the caller's responsibility */
+	window = net_buf_alloc_len(&fragments, 0, K_NO_WAIT);
+	if (!window) {
+		return window;
+	}
+
+	window = bt_buf_make_view(window, outside,
+				  winsize, &get_frag_md(window)->view_meta);
+
+	LOG_DBG("get-acl-frag: outside %p window %p size %d", outside, window, winsize);
+
+	return window;
+}
+#else /* !CONFIG_BT_CONN_TX */
+static struct net_buf *get_data_frag(struct net_buf *outside, size_t winsize)
+{
+	ARG_UNUSED(outside);
+	ARG_UNUSED(winsize);
+
+	/* This will never get called. It's only to allow compilation to take
+	 * place and the later linker stage to remove this implementation.
+	 */
+
+	return NULL;
+}
+#endif /* CONFIG_BT_CONN_TX */
```

Comment:
Doesn't need a change, but a simple question:

Do you generally prefer 
```c
#if COND
static void foo()
{
  bar();
}
#else
static void foo()
{
   baz();
}
#endif
```
over 
```c
static void foo()
{
#if COND
  bar();
#else
   baz();
#endif
}
```

Personally I might lean towards the latter as that has less code duplication, higher chance of working with `IS_ENABLED` and may help intellisense tools to better find the function, as there is only one definition.

Reply:
the latter. Ideally we would have smaller conditionally-compiled files, but C is dum-dum if we don't enable LTO.